test/dynamo/test_decorators.py

```diff
@@ -441,6 +441,105 @@ def fn(x, y):
         res = opt_fn(x, y)
         self.assertEqual(ref, res)
 
+    def test_nonstrict_trace_pre_existing_register_constant_type(self):
+        class State:
+            def get_num(self):
+                torch._dynamo.graph_break()
+                return 42
```

Comment:
@zou3519 I allowed default `__eq__`, because I thought if the object should be id-matched, it'll be a bit odd to force the user to implement a trivial non-default `__eq__` which behaves like a default one.

No strong preference though, mainly because I don't have datapoints to decide the right default here...

Reply:
It's easier to start strict (require a non-trivial eq) and then then relax it later. So I'd prefer that we require a non-trivial eq first just so that we don't need to migrate people over if we end up deciding this is the right thing to do.

Reply:
That's a good call, I'll make it strict.